source/base/utilities.cc

```diff
@@ -132,7 +132,7 @@ namespace Utilities
   trim(const std::string &input)
   {
     std::string::size_type left = 0;
-    std::string::size_type right = input.size() - 1;
+    std::string::size_type right = input.size() > 0 ? input.size() - 1 : 0;
```

Comment:
I never noticed that; it looks like things might explode if one tries to trim an empty string.